### DISCUSSION TITLE
memory usage is very high

### DISCUSSION THREAD

**Discussion by msaoudallah** [mindee/doctr#1208] [General]
hello everyone , i have one thing to discuss with you and lets see if you faced similar situation before. i use doctr for ocr tasks. but i found that there is huge memory increase each time i send a request to my server. note that i have a limited resources and memory leak may lead to server crash or stop.  i went to investigate this and found that one step is taking too much memory and it is not released after request end. the step is when i try to infer the result from a pdf file 
``` 
results = model(doc)
```
result object size depends on the pdf itself but see numbers from 200MB and above which is huge and as i said before they're not released when request completes 
i tred some solutions as follows

- deleting the var by `del` keyword but it appeard to do nothing , 
- also i called `gc.collect()` explicitly and also nothing happened
- combinations of the above steps, memory is released at random but not compared to memory consumed
- creating a subprocess that should clear memory after if finished executing but i'm having issues with this [#1207](https://github.com/mindee/doctr/issues/1207)

do you have any other ideas to share ?

**Comment by Srini138** [comment]
@msaoudallah  does the issue has been resolved for you I'm also facing the same issue could you please provide some insights on how it's consuming very high memory in server little breifly .how did u check that

**Comment by mara004** [comment]
Not sure about possible memory leaks in doctr rsp. its libs.
However, I think the generally high memory consumption can at least partly be explained by doctr using lists where one should use generators. See also #1000, https://github.com/pypdfium2-team/pypdfium2/issues/141 and related threads.

**Comment by TomekPro** [comment]
@msaoudallah @Srini138 have you solved this issue? I'm facing the same problem. Those linesare causingthe problem: (actual running predictions for detection and recognition models)
https://github.com/mindee/doctr/blob/5105f98bcbe736989c25627c12ad71475e1672fd/doctr/models/predictor/pytorch.py#L76
https://github.com/mindee/doctr/blob/5105f98bcbe736989c25627c12ad71475e1672fd/doctr/models/predictor/pytorch.py#L122
